src/Compute/Compute/VirtualMachine/Operation/NewAzureVMCommand.cs

```diff
@@ -966,5 +1036,128 @@ private static string GetStorageAccountNameFromUriString(string uriStr)
+                        writer.WriteLine(keypair.PrivateKey);
+                    }
+                    Console.WriteLine("Private key is saved to " + privateKeyFilePath);
+
+                    using (StreamWriter writer = new StreamWriter(publicKeyFilePath))
+                    {
+                        writer.WriteLine(keypair.PublicKey);
+                    }
+                    Console.WriteLine("Public key is saved to " + publicKeyFilePath);
+
+                    return keypair.PublicKey;
+                }
+
+                throw new Exception("The SSH Public Key resource with name '-SshKeyName' already exists. Either remove '-GenerateSshKey' to use the existing resource or update '-SshKeyName' to create a SSH Public Key resource with a different name.");
+
+            }
+            else
+            {
+                try
+                {
+                    SshPublicKey = this.ComputeClient.ComputeManagementClient.SshPublicKeys.Get(this.ResourceGroupName, this.SshKeyName);
+                    publicKey = SshPublicKey.PublicKey;
+                }
+                catch (Rest.Azure.CloudException)
+                {
+                    throw new Exception("The SSH Public Key resource with name '-SshKeyName' was not found. Either provide '-GenerateSshKey' to create the resource or provide a '-SshKeyName' that exists in the provided Resource Group.");
+                }
+
+                return publicKey;
+            }
+        }
+
+        private VirtualMachine addSshPublicKey(VirtualMachine parameters)
+        {
+            SshPublicKey sshPublicKey = createPublicKeyObject(parameters.OsProfile.AdminUsername);
+            List<SshPublicKey> sshPublicKeys = new List<SshPublicKey>()
+            {
+                sshPublicKey
+            };
+            if (parameters.OsProfile.LinuxConfiguration.Ssh == null)
+            {
+                SshConfiguration sshConfig = new SshConfiguration();
+                parameters.OsProfile.LinuxConfiguration.Ssh = sshConfig;
+            }
+            parameters.OsProfile.LinuxConfiguration.Ssh.PublicKeys = sshPublicKeys;
+
+            return parameters;
+        }
+
+        private void cleanUp()
+        {
+            if (this.GenerateSshKey.IsPresent)
+            {
+                //delete the created ssh key resource
+                WriteInformation("VM creation failed. Deleting the SSH key resource that was created.", new string[] { "PSHOST" });
+                this.ComputeClient.ComputeManagementClient.SshPublicKeys.Delete(this.ResourceGroupName, this.SshKeyName);
+            }
+        }
+
+        private void validate()
+        {
+            if (this.GenerateSshKey.IsPresent && !this.IsParameterBound(c => c.SshKeyName))
+            {
+                throw new Exception("Please provide parameter '-SshKeyName' to be used with '-GenerateSshKey'");
+            }
+
+            if (this.ParameterSetName == "DefaultParameterSet" && !IsLinuxOs())
```

Comment:
is this condition correct?

Reply:
it was faulty. fixed it. thank you for pointing that out